internal/test_cases/history_test_case.go

```diff
@@ -0,0 +1,113 @@
+package test_cases
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/codecrafters-io/shell-tester/internal/assertions"
+	"github.com/codecrafters-io/shell-tester/internal/logged_shell_asserter"
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+// CommandOutputPair represents a command and its expected output
+type CommandOutputPair struct {
+	Command        string
+	ExpectedOutput string
+}
+
+// HistoryTestCase is a test case that:
+// Sends the history command to the shell
+// Verifies that the command is printed to the screen
+// Verifies that the history output shows the command in the expected format
+type HistoryTestCase struct {
+	// SuccessMessage is the message to log in case of success
+	SuccessMessage string
+
+	// CommandsBeforeHistory is a list of commands to execute before running history
+	CommandsBeforeHistory []CommandOutputPair
```

Comment:
Maybe we could just accept `CommandResponseTestCase`s here? That way this supports everything that supports, and can still read `Command` to construct the history assertion